conda_forge_tick/git_utils.py

```diff
@@ -292,41 +372,73 @@ def clone_repo(self, origin_url: str, target_dir: Path):
 
     @lock_git_operation()
     def push_to_url(self, git_dir: Path, remote_url: str, branch: str):
+        """Push changes to a remote URL.
+
+        Parameters
+        ----------
+        git_dir
+            The directory of the git repository.
+        remote_url
+            The URL of the remote.
+        branch
+            The branch to push to.
+
+        Raises
+        ------
+        GitCliError
+            If the git command fails.
         """
-        Push changes to a remote URL.
-        :param git_dir: The directory of the git repository.
-        :param remote_url: The URL of the remote.
-        :param branch: The branch to push to.
-        :raises GitCliError: If the git command fails.
-        """
-
-        self._run_git_command(["push", remote_url, branch], git_dir)
+        try:
+            self._run_git_command(["push", remote_url, branch], git_dir)
+        except GitCliError as e:
+            raise GitCliError("git push failed") from e
```

Comment:
Same here.

Reply:
This change is required because of the new `DOC501` and `DOC502` ruff rules. It doesn't change any behaviour, just makes existing errors more explicit.